console_conf/ui/views/chooser.py

```diff
@@ -47,23 +47,36 @@
 log = logging.getLogger("console_conf.views.chooser")
 
 
-class ChooserCurrentSystemView(BaseView):
+class ChooserBaseView(BaseView):
     title = "Ubuntu Core"
 
-    def __init__(self, controller, current, has_more=False):
-        fmt = "Select one of available actions for \"{}\" by \"{}\"{}."
-        maybe_more = " or view all available systems" if has_more else ""
-        excerpt = fmt.format(current.model.display_name,
-                             current.brand.display_name,
-                             maybe_more)
+    def __init__(self, current, scr):
+        if current is not None:
+            self.title = current.model.display_name
+
+        super().__init__(scr)
+
+
+def by_preferred_action_type(action):
+    """Order action entries by having the 'run' mode first, then 'recover', then
+    'install', the rest is ordered alphabetically."""
+    return (action.mode != "run",
```

Comment:
I find this key function unusual as it returns a tuple (bool, bool, bool, str), I was expecting a (int, str)

```
priority = {"run": 0, "recover": 1, "install": 2}
return (priority.get(action.mode, 100), action.title.lower())
```

Reply:
Nice! I'll push that